src/Serializer/AbstractItemNormalizer.php

```diff
@@ -729,122 +732,146 @@ private function createAttributeValue(string $attribute, mixed $value, string $f
     private function createAndValidateAttributeValue(string $attribute, mixed $value, string $format = null, array $context = []): mixed
     {
         $propertyMetadata = $this->propertyMetadataFactory->create($context['resource_class'], $attribute, $this->getFactoryOptions($context));
-        $type = $propertyMetadata->getBuiltinTypes()[0] ?? null;
+        $types = $propertyMetadata->getBuiltinTypes() ?? [];
+        $isMultipleTypes = \count($types) > 1;
 
-        if (null === $type) {
-            // No type provided, blindly return the value
-            return $value;
-        }
+        foreach ($types as $type) {
+            if (null === $value && $type->isNullable()) {
+                return $value;
+            }
 
-        if (null === $value && $type->isNullable()) {
-            return $value;
-        }
+            $collectionValueType = $type->getCollectionValueTypes()[0] ?? null;
 
-        $collectionValueType = $type->getCollectionValueTypes()[0] ?? null;
+            /* From @see AbstractObjectNormalizer::validateAndDenormalize() */
+            // Fix a collection that contains the only one element
+            // This is special to xml format only
+            if ('xml' === $format && null !== $collectionValueType && (!\is_array($value) || !\is_int(key($value)))) {
```

Comment:
@dunglas should we really support this? We discussed XML is not supporte-able by our normalizers at the symfony live. Maybe that deprecating the support of XML (w/ JSON by default) for 3.2 should be our target?